packages/web-components/src/wallet-connection/watchWallet.js

```diff
@@ -227,9 +227,11 @@ export const watchWallet = (chainStorageWatcher, address, rpc) => {
       unserializer: chainStorageWatcher.marshaller,
     });
 
-    for await (const { value } of iterateLatest(follower)) {
-      console.debug('wallet update', value);
-      walletUpdatesNotifierKit.updater.updateState(harden(value));
+    for await (const update of iterateEach(follower)) {
+      console.debug('wallet update', update);
+      if (!('error' in update)) {
+        walletUpdatesNotifierKit.updater.updateState(harden(update.value));
+      }
```

Comment:
consider not inverting the condition.
```suggestion
      if ('error' in update) continue;
      walletUpdatesNotifierKit.updater.updateState(harden(update.value));
```